### PR TITLE
add libm support

### PR DESCRIPTION
**Pull request by stlankes**
libm is port of MUSL's libm to Rust. This crate is used to support basic arithmetic functions.

### CODE REVIEW

File: demo/src/tests/mod.rs [hunk 1]
```diff
@@ -275,6 +275,18 @@ pub fn hello() -> Result<(), ()> {
...
+
+pub fn arithmetic() -> Result<(), ()> {
+	let mut rng = rand::thread_rng();
+	let x = rng.gen::<f64>();
+	let y: f64 = x.exp();
+	let z: f64 = y.log(E);
+
+	println!("x = {}, e^x = {}, ln(e^x) = {}", x, y, z);
+
 	Ok(())
```
Comment (jschwe) [inline demo/src/tests/mod.rs:290]: Maybe add a check here if x and z are approximately the same instead of always returning `Ok(())`

Comment (stlankes) [reply demo/src/tests/mod.rs:290]: Good idea

File: hermit-sys/src/cmath.rs [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+pub extern "C" fn atan(n: f64) -> f64 {
+	libm::atan(n)
+}
+
+#[no_mangle]
+pub extern "C" fn atan2(a: f64, b: f64) -> f64 {
+	libm::atan2(a, b)
+}
+
+pub extern "C" fn atan2f(a: f32, b: f32) -> f32 {
```
Comment (jschwe) [inline hermit-sys/src/cmath.rs:53]: I think `#[no_mangle]` is missing here

File: hermit-sys/src/cmath.rs [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+pub extern "C" fn hypotf(x: f32, y: f32) -> f32 {
+	libm::hypotf(x, y)
+}
+
+#[no_mangle]
+pub extern "C" fn log(x: f64) -> f64 {
+	libm::log(x)
+}
+
+pub fn logf(x: f32) -> f32 {
```
Comment (jschwe) [inline hermit-sys/src/cmath.rs:147]: Is this one intentionally a rust function?

File: hermit-sys/src/cmath.rs (hunk 2 above)
Comment (stlankes) [reply hermit-sys/src/cmath.rs:54]: Yes, I revise the PR

**Pull request merged**